[RAPTOR-10015] rearrange requirements files

yakov-g opened the pull request:
# This repository is public. Do not put here any private DataRobot or customer's data: code, datasets, model artifacts, .etc.

## Summary

- Add requirements_test_functional.txt and move dr-usertool there.
- remove unused

## Rationale

yakov-g commented:
jenkins run test_integration_java_environment

yakov-g commented:
jenkins run pipelineController:all-expected-tasks

zohar-mizrahi reviewed (commented):
You decided to provide both `requirements_test_functional.txt` and `requirements_test.txt`. What is the purpose of it? 
Does this mean that whoever works on development doesn't need the functional tests?

yakov-g commented:
jenkins run pipelineController:all-expected-tasks

eric-s-s commented on README.md:
```diff
@@ -116,10 +116,14 @@ Rscript -e 'library(caret); install.packages(unique(modelLookup()[modelLookup()$
-1. `pip install -r requirements_dev.txt`
-1. `pip install -e custom_model_runner/`
-1. pytest to your heart's content
+1. Create Py venv, >=3.9 is recommended.
```
```suggestion
1. Create virtualenv  Python>=3.9
```
is this true?  i think we have conflicts somewhere with higher versions.

yakov-g replied:
done

yakov-g commented:
> You decided to provide both `requirements_test_functional.txt` and `requirements_test.txt`. What is the purpose of it? Does this mean that whoever works on development doesn't need the functional tests?

I removed requiments_dev, and updated README.
I can not  think of proper instructions regarding functional tests right now.

yakov-g commented:
jenkins run test_integration_general test_integration_r_environment

yakov-g commented:
jenkins run test_integration_general
